### PR TITLE
Review/elixirconfeu19

### PR DESCRIPTION
**Pull request by gemantzu**

### CODE REVIEW

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+Forum hall is a huge atrium hall, which could easily house all of us in one sitting. It had great visibility of both the speaker and his notes. It also had great acoustics.
+
+South hall on the other hand, was a step down. It was much smaller, with three medium monitors where you could view the speakers’ notes. Sadly, it also had some pillars installed inside, and it was flat, which made viewing both the speaker and his notes a challenge.
+
+From the beginning of the conference, I got to see some familiar faces from last year, which made me much more comfortable. I think the conference was more crowded this year, but we have to check with the organizers to have official stats on the attendees.
+
+I got to talk with all of the sponsors that had a booth in the venue, which was really fun. Check out Cultivate, solarisBank, coders51, bitcrowd, ScoutAPM, Erlang Solutions and Toyota Connected, they are doing some great stuff with Elixir and some of them are hiring as well. They were kind enough to provide us with some swag, including some very nice t-shirts and a ton of stickers! Check it out:
```
Comment (tajchumber) [inline _posts/2019-06-06-elixir-conf-eu-2019-review.md]: "checked out" doesn't sound right maybe use "I explored"

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+We were shown how the library gets to be lightweight both on the server and on the client, leveraging the power of Elixir, and also, how every component is supervised and can be brought back up upon crashing.
+
+We also learned that the library is not going to affect SEO, as the initial load of the component is treated as standard HTML. It is also really easy to test LiveView and it is really not necessary to do testing with headless browser launching etc.
+
+[![Phoenix Live View](https://img.youtube.com/vi/8xJzHq8ru0M/0.jpg)](https://www.youtube.com/watch?v=8xJzHq8ru0M)
+
+### Live coding an Escher painting using Scenic by Ju Liu
+
+Ju Liu gave a very good talk about how to draw a simplified version of Escher's fish painting using vectors, a fish image and scenic. The talk  had not much to do with Scenic––it was more about Vector theory and how to use small graphic assets to compose more complex structures.
```
Comment (tajchumber) [inline _posts/2019-06-06-elixir-conf-eu-2019-review.md]: "The talk  had": there are two spaces between "talk" and "had"

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+### Rewriting critical software in Elixir - a case study by Renan Ranelli
+
+Renan Ranelli gave us some very insightful information on how to rewrite critical software in Elixir while the old system is still a moving target. Renan works in Telnyx, an IP Telephony company, where their old python system was slowing down in development due to tech debt.
+
+He began giving us some small insight on how IP telephony works, and the functionality of their Dialplan service that was up for a rewrite. That service is almost stateless, super latency sensitive and had relatively low throughput for each server. The service produced an xml file, with all the information the switch needs to to operate on a call. 
+
+The first step in the case study was geting buy-in from stakeholders by convincing them that the company needs to slow down feature development to pay technical debt, which they accomplished phrasing all the benefits in business terms. The next step was to commit resources, which is very critical as a rewrite is a huge amount of work when you are chasing a moving target. Another step is to write and deploy continuously.
```
Comment (tajchumber) [inline _posts/2019-06-06-elixir-conf-eu-2019-review.md]: typo "geting" should be "getting"

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+The 3 lessons they have learned, are the following:
+
+1) Let Flow do the work
+2) Organize your Flow
+3) Tune your Flow
+
+In Lesson 1, we learn about GenStage / Flow and how their amazing features can help with a big long running system like that. Parallelism is made easy, and by using Flow, you are protected from DDoSing yourself, because instead of pushing from one system to another via stream, now you pull as many messages as you can get.
+
+In Lesson 2, John talked about how one should know their system pretty well, so that they can provide the correct configuration to it. For example, by using SQS, you are limited to receiving batches of 10 messages each time, so you have to build around that, unless you want to wait for say 500 messages to arrive at your system, 10 at a time. Another useful tip is to use a token, meaning a struct that gets pushed down through the whole system, changed at each step, which makes processing and pattern matching on messages really easy (it’s the same thing that Plug uses with it’s Conn struct). Errors are supposed to be passed through the whole flow, and left out of acknowledgement, so SQS will resend them, and you can process them again.
```
Comment (tajchumber) [inline _posts/2019-06-06-elixir-conf-eu-2019-review.md]: "...unless you want to wait for say 500 messages to arrive at your system, 10 at a time."
Maybe it's me but this doesn't make sense to me. Can you explain please?

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+[![flow](https://img.youtube.com/vi/t46L9RKmlNo/0.jpg)](https://www.youtube.com/watch?v=t46L9RKmlNo)
+
+### Telemetry ...and metrics for all - ElixirConf EU 2019 by Arkadiusz Gil
+
+Arkadiusz Gil's talk was about Telemetry, a dynamic dispatching library used for metrics and instrumentations.
+
+He started by talking about monitoring, and how crucial it is to improving performance. The goal is to use monitoring early on in the lifecycle of our project (as we have done already with testing and deployment), so we can use it as another form of verification. Next, he showed us how we can do metrics today, using custom functions (like the number of requests, the number of successful responses, how big a load we are pushing to external systems like the the database, how to track memory or cpu usage from the vm etc). 
+
+But as we add more and more, this custom set up becomes repetitive. Telemetry uses emitted events from various parts of our systems, and then attaches to those events by using handlers, which are in turn pushed to a reporter system like statsd.
```
Comment (tajchumber) [inline _posts/2019-06-06-elixir-conf-eu-2019-review.md]: "emitted events" should be "events emitted"?

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+Lastly, he demonstrated the way you can plug Telemetry into a project by using the Rumble application from the Phoenix Programming book. It is as simple as adding the dependencies in `mix.exs`, making a custom module that defines which metrics we want, supervising it in `application.ex` and we are good to go from the application standpoint. The difficult part is setting up the reporter, like statsd, but there is going to be one included in Phoenix as well, where we will be able to access it from a route.
+
+[![](https://img.youtube.com/vi/cOuyOmcDV1U/0.jpg)](https://www.youtube.com/watch?v=cOuyOmcDV1U)
+
+## Erlang Ecosystem Foundation
+
+As they did in CODE BEAM SF conference, we were given the presentation on the new [Erlang Ecosystem Foundation](https://erlef.org/). They stated again that things are still a little blurry, and we will probably have more information in the coming months, but check out the video of the presentation just to know what's coming.
```
Comment (tajchumber) [inline _posts/2019-06-06-elixir-conf-eu-2019-review.md]: should you add the yt video of their presentation here?

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md (hunk 1 above)
Comment (gemantzu) [reply _posts/2019-06-06-elixir-conf-eu-2019-review.md]: I don't have `checked out` spelled anywhere. Do you mean on line 36 `Check out`? I am proposing people to lookup those companies and their work. Do you think I should phrase it differentely?

Comment (tajchumber) [reply _posts/2019-06-06-elixir-conf-eu-2019-review.md]: @gemantzu Yes I meant `Check out` sorry.
If that is your meaning, than it's absolutely fine, I think I misunderstood the meaning of the sentence :)

File: _posts/2019-06-06-elixir-conf-eu-2019-review.md (hunk 6 above)
Comment (gemantzu) [reply _posts/2019-06-06-elixir-conf-eu-2019-review.md]: It hasn't been uploaded yet. In general, they have taken their time uploading the videos for that conference, Idk why.

**Pull request merged**